### PR TITLE
Simplified expressions

### PR DESCRIPTION
**Pull request by GYuvanShankar**
## Summary of Changes

Simplified a few expressions

## Release Management

* Affected package(s): Principal_component_analysis

### CODE REVIEW

File: Principal_component_analysis/include/CGAL/linear_least_squares_fitting_circles_2.h [hunk 1]
```diff
@@ -194,7 +194,7 @@ linear_least_squares_fitting_2(InputIterator first,
                    0.0, radius};
     Matrix transformation = init_matrix<FT>(2,delta);
     FT length = 2 * radius;
-    CGAL_assertion(length != 0.0);
+    CGAL_assertion(length != FT(0));
```
Comment (mglisse) [inline Principal_component_analysis/include/CGAL/linear_least_squares_fitting_circles_2.h:197]: If we are going to change it, we might as well use `CGAL::is_zero` I guess?

Comment (GYuvanShankar) [reply Principal_component_analysis/include/CGAL/linear_least_squares_fitting_circles_2.h]: Done.

File: Principal_component_analysis/include/CGAL/linear_least_squares_fitting_circles_2.h [hunk 2]
```diff
@@ -111,9 +111,9 @@ linear_least_squares_fitting_2(InputIterator first,
 
   // Translate the 2nd order moment calculated about the origin to
   // the center of mass to get the covariance.
-  covariance[0] += -mass * (c.x() * c.x());
-  covariance[1] += -mass * (c.x() * c.y());
-  covariance[2] += -mass * (c.y() * c.y());
+  covariance[0] -= mass * (c.x() * c.x());
```
Comment (afabri) [inline Principal_component_analysis/include/CGAL/linear_least_squares_fitting_circles_2.h:114]: While you are at it, we also have `CGAL::square()` which may be faster depending on the number type, so here ot could be      `square(c.x())`.

File: Principal_component_analysis/include/CGAL/linear_least_squares_fitting_segments_2.h [hunk 3]
```diff
@@ -104,9 +104,9 @@ linear_least_squares_fitting_2(InputIterator first,
 
   // Translate the 2nd order moment calculated about the origin to
   // the center of mass to get the covariance.
-  covariance[0] += -mass * ( c.x() * c.x());
-  covariance[1] += -mass * (c.x() * c.y());
-  covariance[2] += -mass * (c.y() * c.y());
+  covariance[0] -= mass * CGAL::square(c.x()));
```
Comment (afabri) [inline Principal_component_analysis/include/CGAL/linear_least_squares_fitting_segments_2.h:107]: One parenthesis too much.    As soon as a PR is marked for Under Testing this means that it got integrated in the testsuite.   There you can see that the line of Principal Component Analysis is fully red.     It is best practice to compile and run the trstsuite of a package that gets modifed locally before pushing.

Comment (sloriot) [inline Principal_component_analysis/include/CGAL/linear_least_squares_fitting_segments_2.h:107]:
```suggestion
  covariance[0] -= mass * CGAL::square(c.x());
```

Comment (GYuvanShankar) [reply Principal_component_analysis/include/CGAL/linear_least_squares_fitting_segments_2.h]: My bad. Will follow that procedure.

**Comment by sloriot** [comment]
Successfully tested in CGAL-5.5-Ic-10

**Pull request merged**